src/serverenvironment.cpp

```diff
@@ -1753,14 +1753,24 @@ void ServerEnvironment::getSelectedActiveObjects(
 			continue;
 
 		v3f pos = obj->getBasePosition();
-
-		aabb3f offsetted_box(selection_box.MinEdge + pos,
-			selection_box.MaxEdge + pos);
+		v3f rel_pos = shootline_on_map.start - pos;
 
 		v3f current_intersection;
-		v3s16 current_normal;
-		if (boxLineCollision(offsetted_box, shootline_on_map.start, line_vector,
-				&current_intersection, &current_normal)) {
+		v3f current_normal;
+
+		ObjectProperties *props = obj->accessObjectProperties();
+		bool rotate_selectionbox = props ? props->rotate_selectionbox : false;
+		bool collision;
+		UnitSAO* usao = dynamic_cast<UnitSAO*>(obj);
```

Comment:
Downcast to a UnitSAO is needed to access the rotation; the alternative would be to move the rotation to the base SAO class, or to even merge the two classes (as it appears all SAOs are UnitSAOs). Again I consider such a refactor out of scope for this PR. Perhaps a comment should be added though?